File: frontend/src/types/attribute.ts
export enum AttributeOwnerType {
  SHIFT = 0,
  WORKER = 1,
}

export type AttributeT = {
  id: string;
  value: string | number | boolean;
  ownerType: AttributeOwnerType;
  ownerId: string;
  dimensionId: string;
  dimEntryIds: string[];
};

/**
 * Convert API response to AttributeT
 */
export function toAttributeT(data: any): AttributeT {
  return {
    id: data.id,
    value: data.value,
    ownerType: data.owner_type ?? data.ownerType,
    ownerId: data.owner_id ?? data.ownerId,
    dimensionId: data.dimension_id ?? data.dimensionId,
    dimEntryIds: data.dim_entry_ids ?? data.dimEntryIds ?? [],
  };
}

File: backend/src/core/attribute.py
from dataclasses import dataclass
from enum import Enum
from typing import List


class AttributeOwnerType(Enum):
    SHIFT = 0
    WORKER = 1


@dataclass
class Attribute:
    id: str
    value: str | int | float | bool
    owner_type: AttributeOwnerType
    owner_id: str
    dimension_id: str
    dim_entry_ids: List[str]
